File: apps/api/app/schemas/health.py
"""Health Check Pydantic Schemas."""

from __future__ import annotations

from pydantic import BaseModel, Field


class HealthResponse(BaseModel):
    """Structured response for health check endpoint."""

    status: str = Field(..., description="Overall service status ('healthy', 'degraded')")
    database: str = Field(..., description="Database connectivity status ('connected', 'disconnected')")
    environment: str = Field("development", description="Runtime environment")
    version: str = Field("0.1.0", description="API Version")

File: apps/web/src/lib/api/health.ts
/**
 * Service & Database Health Check API Endpoint
 */

import { fetchJson } from './client';
import type { HealthResponse } from './types';

export async function getHealthStatus(): Promise<HealthResponse> {
  return fetchJson<HealthResponse>('/health', {
    method: 'GET',
    timeoutMs: 5000,
  });
}
